src/storage/store/engine.rs

```diff
@@ -444,11 +444,17 @@ impl ShardEngine {
                         "Fname transfer has no proof"
                     );
                 }
-                let proof = fname_transfer.proof.as_ref().unwrap();
-                // TODO: Verify the EIP-712 server signature
+
+                match fname_transfer.verify_signature() {
```

Comment:
We should also do the same thing here https://github.com/farcasterxyz/snapchain-v0/blob/f5f1aff56fc11f9fb8421e6eeab9277628e35998/src/storage/store/engine.rs#L853-L855 for username proof claim signatures (I just looked and looks like we never call `verifyUserNameProofClaim` in hubs, this might be a bug?) and the eth verified addresses.

Could you also implement sol verified addresses (maybe in a separate PR? upto you) before moving on to the other validations?